lib/ts/recipe/passwordless/index.ts

```diff
@@ -92,9 +136,56 @@ export default class RecipeWrapper {
           }
         | { status: "RESTART_FLOW_ERROR"; fetchResponse: Response }
     > {
-        return Recipe.getInstanceOrThrow().recipeImplementation.consumeCode({
-            ...input,
-            userContext: getNormalisedUserContext(input.userContext),
+        const recipe: Recipe = Recipe.getInstanceOrThrow();
+        const normalisedUserContext = getNormalisedUserContext(input.userContext);
+
+        const attemptInfoFromStorage = await recipe.recipeImplementation.getLoginAttemptInfo({
+            userContext: normalisedUserContext,
+        });
+
+        if (attemptInfoFromStorage === undefined) {
+            throw new Error(
```

Comment:
If the user clicks on a magic link and then logs out, and then clicks on the same link again, we will see this error. However, we should return `RESTART_FLOW_ERROR`? That can be done by just setting these as empty strings and then calling the API?

Reply:
To keep the logic consistent with resend code, if the method is called with a `userInputCode` and `deviceId` then we default to empty strings if the `attemptInfoFromStorage` is undefined (resulting in a `RESTART_FLOW_ERROR`). 

If the method is called with a link code then both `linkCode` and `preAuthSessionId` are fetched from the query params which default to empty strings as well